app/views/devise/registrations/new.html.erb

```diff
@@ -26,6 +36,16 @@
     <%= f.password_field :password_confirmation, autocomplete: "off" %>
   </div>
 
+   <div class="field">
+    <%= f.label :profile_photo %><br />
+    <%= f.file_field :profile_photo, autofocus: true, autocomplete: "Profile Photo" %>
+  </div>
+
+   <div class="field">
+    <%= f.label :interests %><br />
+    <%= f.text_field :interests, autofocus: true, autocomplete: "Interests" %>
+  </div>
+
```

Comment:
When creating new users, need a conditional to prevent users from having the same username. Right now, it is possible to have two users with the exact same username. It is also possible to sign up with an empty value for username - believe this should be required.

There seems to be something wrong with the user sessions - note that user id#2 has a sign in count of 5, so clearly my attempts to sign in worked, but I wasn't able to stay signed in after the error on redirect

![screen shot 2018-08-20 at 8 49 46 pm](https://user-images.githubusercontent.com/36312228/44375803-0d132c00-a4bb-11e8-88db-76984cce8bb4.png)

Reply:
Good call. Added the following code to models/user.rb to ensure:
1. Username needed to be unique
2. Username needs a value
```
class User < ApplicationRecord
  # Include default devise modules. Others available are:
  # :confirmable, :lockable, :timeoutable and :omniauthable
  devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :trackable, :validatable, :confirmable
  validates :user_name, uniqueness: true, presence: true

end

```

Reply:
Username uniqueness and presence are now both validated, so that looks solid to me. Am currently checking on the redirect error.